File: practice18/3.cpp
//https://www.nowcoder.com/practice/0425aa0df74646209d3f56f627298ab2?tpId=98&tqId=32845&tPage=2&rp=2&ru=/ta/2019test&qru=/ta/2019test/question-ranking
#include<iostream>
#include<string>
#include<algorithm>
using namespace std;
int main()
{
	string str;
	int m, n, i, j, x;
	while (cin >> n)
	{
		string *pt = new string[n];
		string *pt1 = new string[n];
		for (i = 0; i < n; i++)
		{
			cin >> pt[i];
		}
		for (i = 0; i < n; i++)
		{
			pt1[i] = "000000";
			m = pt[i].size();
			x = m - 6;
			copy(pt[i].begin() + x, pt[i].end(), pt1[i].begin());
		}
		for (i = 0; i < n - 1; i++)
		{
			for (j = i + 1; j < n; j++)
			{
				if (pt1[i] > pt1[j])
				{
					str = pt1[i], pt1[i] = pt1[j], pt1[j] = str;
				}
			}
		}
		for (i = 0; i < n; i++)
		{
			cout << pt1[i] << endl;
		}
	}
	system("pause");
	return 0;
}

File: practice28/3.java


//https://www.nowcoder.com/practice/f350f14cd22c41aabfa7e54a1b8e8825?tpId=98&tqId=33024&tPage=11&rp=11&ru=/ta/2019test&qru=/ta/2019test/question-ranking
import java.io.*;
import java.util.*;

public class Main {
    public static void translate(String s){
        String []str=s.split(",");
        List<Integer> pt=new LinkedList<>();
        for(int i=0;i<str.length;i++){
            pt.add(Integer.parseInt(str[i]));
        }
        ListIterator<Integer> front=pt.listIterator();
        ListIterator<Integer> back=pt.listIterator(pt.size());

        List<Integer> pt1=new LinkedList<>();
        for(int i=0;i<str.length/2;i++){
            pt1.add(front.next());
            pt1.add(back.previous());
        }
        if(str.length%2==1){
            pt1.add(front.next());
        }
        front=pt1.listIterator();
        System.out.print(front.next());
        while(front.hasNext()){
            System.out.print(","+front.next());
        }
        System.out.println();
    }
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader=new BufferedReader(new InputStreamReader(System.in));
        String s=bufferedReader.readLine();
        translate(s);
    }
}